### PR TITLE
clang12: Add missing GNU headers include path

### LINKED ISSUE #9198: `clang` does not detect GNU extensions in Haiku
Using clang 12 on Haiku `hrev57210`, with this source file:

```c
#include <math.h>
#include <stdint.h>

int main()
{
	return (int)(intptr_t)(void*)&sincos;
}
```

Compilation using `gcc test.c -o test -D_GNU_SOURCE` works normally.

However, compilation using `clang` fails:

```
~/Desktop> clang test.c -o test -D_GNU_SOURCE
test.c:6:32: error: use of undeclared identifier 'sincos'
        return (int)(intptr_t)(void*)&sincos;
                                      ^
1 error generated.
```

### PR DESCRIPTION
**Pull request by trungnt2910**
Should fix #9198.

### CODE REVIEW

**Comment by trungnt2910** [comment]
Added the Python changes from #9195 as requested by @Begasus on IRC.

File: sys-devel/llvm/patches/clang-12.0.1.patchset [hunk 1]
```diff
@@ -194,3 +194,28 @@ index 50a20c2..ac65381 100644
...
+--- a/lib/Frontend/InitHeaderSearch.cpp
++++ b/lib/Frontend/InitHeaderSearch.cpp
+@@ -331,6 +331,7 @@ void InitHeaderSearch::AddDefaultCIncludePaths(const llvm::Triple &triple,
+     AddPath("/boot/system/develop/headers/3rdparty", System, false);
+     AddPath("/boot/system/develop/headers/bsd", System, false);
+     AddPath("/boot/system/develop/headers/glibc", System, false);
++    AddPath("/boot/system/develop/headers/gnu", System, false);
+     AddPath("/boot/system/develop/headers/posix", System, false);
+ #ifdef HAIKU_HYBRID_SECONDARY
+     AddPath("/boot/system/develop/headers/" HAIKU_HYBRID_SECONDARY, System, false);
```
Comment (jmairboeck) [inline sys-devel/llvm/patches/clang-12.0.1.patchset:218]: Did you miss the same for `HAIKU_HYBRID_SECONDARY`? It is not visible from this patch, but I suppose that list is supposed to contain the same directories a second time for the secondary architecture.

Comment (trungnt2910) [reply sys-devel/llvm/patches/clang-12.0.1.patchset:218]: I don't think so. `headers/bsd` is only added in that one place, so `headers/gnu` should be treated similarly.

Comment (jmairboeck) [reply sys-devel/llvm/patches/clang-12.0.1.patchset:218]: You are right, these don't exist in the `x86` subdirectory. Sorry for the noise.

**Pull request merged**